### PR TITLE
Refactor wheel_installer

### PR DESCRIPTION
**Pull request by groodt**
Refactors `extract_single_wheel.py` into `wheel_installer.py`. Also relocates some (but not all) of the various methods and functions into this single script. There was a lot of confusing indirection between modules.

The `wheel_installer.py` logic is quite straightforward and there isn't much benefit in splitting it out across so many files. There is more refactoring that can be done, but this was a chunky first-pass.

Also ran the `black` formatter, so there are a few unrelated formatting changes.

### CODE REVIEW

File: examples/wheel/wheel_test.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def test_customized_wheel(self):
 examples/wheel/lib/module_with_data.py,sha256=8s0Khhcqz3yVsBKv2IB5u4l4TMKh7-c_V6p65WVHPms,637
 examples/wheel/lib/simple_module.py,sha256=z2hwciab_XPNIBNH8B1Q5fYgnJvQTeYf0ZQJpY8yLLY,637
 examples/wheel/main.py,sha256=sgg5iWN_9inYBjm6_Zw27hYdmo-l24fA-2rfphT-IlY,909
-""")
+""",
```
Comment (groodt) [inline examples/wheel/wheel_test.py:112]: black formatting

File: python/pip_install/extract_wheels/BUILD [hunk 2]
```diff
@@ -78,18 +77,6 @@ py_test(
     ],
 )
 
-py_test(
```
Comment (groodt) [inline python/pip_install/extract_wheels/BUILD:81]: Moved to wheel_installer_test

File: python/pip_install/extract_wheels/BUILD [hunk 3]
```diff
@@ -126,16 +113,6 @@ py_test(
...
```
Comment (groodt) [inline python/pip_install/extract_wheels/BUILD:129]: Moved to wheel_installer_test

File: python/pip_install/extract_wheels/bazel.py [hunk 4]
```diff
@@ -1,206 +1,10 @@
...
-from python.pip_install.extract_wheels import annotation, namespace_pkgs, wheel
-
 WHEEL_FILE_LABEL = "whl"
 PY_LIBRARY_LABEL = "pkg"
 DATA_LABEL = "data"
 DIST_INFO_LABEL = "dist_info"
 WHEEL_ENTRY_POINT_PREFIX = "rules_python_wheel_entry_point"
 
 
-def generate_entry_point_contents(
```
Comment (groodt) [inline python/pip_install/extract_wheels/bazel.py:18]: Moved to wheel_installer (only callsite)

File: python/pip_install/extract_wheels/bazel_test.py [hunk 5]
```diff
@@ -1,26 +0,0 @@
-import unittest
-
-from python.pip_install.extract_wheels.bazel import generate_entry_point_contents
-
-
-class BazelTestCase(unittest.TestCase):
```
Comment (groodt) [inline python/pip_install/extract_wheels/bazel_test.py:6]: Moved to wheel_installer_test

File: python/pip_install/extract_wheels/extract_single_wheel.py [hunk 6]
```diff
@@ -1,105 +0,0 @@
...
-import os
-import subprocess
-import sys
-from tempfile import NamedTemporaryFile
-
-from python.pip_install.extract_wheels import arguments, bazel, requirements
-from python.pip_install.extract_wheels.annotation import annotation_from_str_path
-
-
-def configure_reproducible_wheels() -> None:
```
Comment (groodt) [inline python/pip_install/extract_wheels/extract_single_wheel.py:13]: Moved to wheel_installer (only callsite)

File: python/pip_install/extract_wheels/requirements.py [hunk 7]
```diff
@@ -1,47 +0,0 @@
-import re
-from typing import Dict, Optional, Set, Tuple
-
-from pip._vendor.packaging.utils import canonicalize_name
-
-
-def parse_extras(requirements_path: str) -> Dict[str, Set[str]]:
```
Comment (groodt) [inline python/pip_install/extract_wheels/requirements.py:7]: Moved to wheel_installer (only callsite)

File: tools/bazel_integration_test/test_runner.py [hunk 8]
```diff
@@ -56,7 +56,7 @@ def main(conf_file):
                 # TODO: --override_module isn't supported in the current BAZEL_VERSION (5.2.0)
                 # This condition and attribute can be removed when bazel is updated for
                 # the rest of rules_python.
-                if (config["bzlmod"]):
+                if config["bzlmod"]:
```
Comment (groodt) [inline tools/bazel_integration_test/test_runner.py:59]: black formatter

File: tools/wheelmaker.py [hunk 9]
```diff
@@ -275,15 +275,15 @@ def parse_args() -> argparse.Namespace:
         action="append",
         default=[],
         help="Path prefix to be stripped from input package files' path. "
-             "Can be supplied multiple times. Evaluated in order.",
+        "Can be supplied multiple times. Evaluated in order.",
```
Comment (groodt) [inline tools/wheelmaker.py:278]: black formatter

**Review by f0rmiga — APPROVED** [review]

**Pull request merged**